app/components/combined_system_message/combined_system_message.js

```diff
@@ -211,6 +213,8 @@ export default class CombinedSystemMessage extends React.PureComponent {
         if (allUsernames.length > 0) {
             this.props.actions.getMissingProfilesByUsernames(allUsernames);
         }
+
+        this.props.actions.getChannelStats(this.props.currentChannelId);
```

Comment:
why fetching the stats on each `combined_system_message` that gets rendered? this does not look right to me

Reply:
@enahum After your suggestions given to me, I was able to pinpoint and found out that this is currently a problem in the web-app ( https://github.com/mattermost/mattermost-server/issues/11823 ) . I have removed this line of code. If users are added manually to a channel or a user joins a channel themself, then the `user_added` websocket action is broadcasted. Thanks for your help!